src/main/java/io/rpg/model/actions/BattleAction.java

```diff
@@ -0,0 +1,28 @@
+package io.rpg.model.actions;
+
+import io.rpg.model.object.GameObject;
+import io.rpg.model.object.Player;
+
+public class BattleAction {
+    private final Player player;
+    private final GameObject opponent;
```

Comment:
It would be easer for parsing if you would replace it with tag of wanted GameObject